django/db/models/constraints.py

```diff
@@ -43,7 +43,14 @@ def get_violation_error_message(self):
     def deconstruct(self):
         path = "%s.%s" % (self.__class__.__module__, self.__class__.__name__)
         path = path.replace("django.db.models.constraints", "django.db.models")
-        return (path, (), {"name": self.name})
+        return (
+            path,
+            (),
+            {
+                "name": self.name,
+                "violation_error_message": self.violation_error_message,
+            },
+        )
```

Comment:
We should add `violation_error_message` only when it's set and when it's different from the default value, e.g.
```python
if self.violation_error_message and self.violation_error_message != _("Constraint “%(name)s” is violated."):
    ...
```
We could also add the `default_violation_error_message` attribute to the class:
```python
class BaseConstraint:
    default_violation_error_message = _("Constraint “%(name)s” is violated.")
    violation_error_message = None

    def __init__(self, name, violation_error_message=None):
        self.name = name
        if violation_error_message is not None:
            self.violation_error_message = violation_error_message
        else:
            self.violation_error_message = self.default_violation_error_message

    def deconstruct(self):
        path = "%s.%s" % (self.__class__.__module__, self.__class__.__name__)
        path = path.replace("django.db.models.constraints", "django.db.models")
        kwargs = {"name": self.name}
        if self.violation_error_message is not None and self.violation_error_message != self.default_violation_error_message:
            kwargs['violation_error_message'] = self.violation_error_message
        return (path, (), kwargs)
```

Reply:
I agree, it would have allowed me to not update all deconstruct tests.
But as we have a mix between a class and an instance attribute, I was not sure how to do this correctly in respect to the existing code or the code rules of the project, so I made the smallest change I could.

Reply:
> I agree, it would have allowed me to not update all deconstruct tests. But as we have a mix between a class and an instance attribute, I was not sure how to do this correctly in respect to the existing code or the code rules of the project, so I made the smallest change I could.
> 
> PS: Your solution is great, I thought about it but was not sure I really could make such a change (renaming the class attr)

IMO, it's acceptable and it should be backward compatible. `violation_error_message` is still in place, we just change a way of initialization.